jf_agent/main.py

```diff
@@ -582,6 +583,12 @@ def download_and_write_issues():
             config.compress_output_files,
             download_customfieldoptions(jira_connection),
         )
+        write_file(
+            config.outdir,
+            'jira_components',
+            config.compress_output_files,
+            download_project_components(jira_connection),
+        )
```

Comment:
Given the coupling between projects:components seems analogous to projects:versions, I had assumed we would follow a strategy similar to `jira_projects_and_versions` where the components are coupled with the projects.

I don't have enough specific agent experience to say why it would be better or worse to do this, just wanted to say that I had assumed there was a need to do things similarly to versions.